src/main/scala/com/metl/model/BackendAdaptor.scala

```diff
@@ -550,3 +551,28 @@ class ResourceCachingAdaptor(sc:ServerConfiguration,cacheConfig:CacheConfig) ext
     super.isReady
   }
 }
+
+class D2LCachingAdaptor(cacheConfig:CacheConfig) {
```

Comment:
How many DB connections are you making during that time, and are they in parallel or series?